packages/gatsby-plugin-page-creator/src/get-collection-route-params.ts

```diff
@@ -25,12 +26,15 @@ export function getCollectionRouteParams(
     }
 
     const key = extractFieldWithoutUnion(part)
-    // Use the previously created regex to match prefix-123 to prefix-(.+)
-    const match = urlParts[i].match(templateRegex[i])
 
-    if (match) {
-      params[key] = match[1]
-    }
+    key.forEach((k, j) => {
+      // Use the previously created regex to match prefix-123 to prefix-(.+)
+      const match = urlParts[i].match(templateRegex[i])
+
+      if (match) {
+        params[k] = match[j + 1]
+      }
+    })
   })
```

Comment:
Would it make sense to shortcut the loop once a match was found? (`key.some((k, j) => {` and returning true when matched).